src/analysis/raster/qgskde.cpp

```diff
@@ -164,13 +176,14 @@ QgsKernelDensityEstimation::Result QgsKernelDensityEstimation::addFeature( const
     unsigned int xPosition = ( ( ( *pointIt ).x() - mBounds.xMinimum() ) / mPixelSize ) - buffer;
     unsigned int yPosition = ( ( ( *pointIt ).y() - mBounds.yMinimum() ) / mPixelSize ) - buffer;
 
+    // calculate buffer around pixel
+    QgsRectangle extent( ( *pointIt ).x() - radius, ( *pointIt ).y() - radius, ( *pointIt ).x() + radius, ( *pointIt ).y() + radius );
+
     // get the data
-    float *dataBuffer = ( float * ) CPLMalloc( sizeof( float ) * blockSize * blockSize );
-    if ( GDALRasterIO( mRasterBandH, GF_Read, xPosition, yPosition, blockSize, blockSize,
-                       dataBuffer, blockSize, blockSize, GDT_Float32, 0, 0 ) != CE_None )
-    {
-      result = RasterIoError;
-    }
+    QgsRasterBlock *block = mProvider->block( 1, extent, blockSize, blockSize );
+    QByteArray blockData = block->data();
+    blockData.detach();
```

Comment:
This detach is probably not necessary as the detach should happen on the line below in data() call if necessary